temporallib/client/client.py

```diff
@@ -50,29 +51,32 @@ def _init_runtime_with_prometheus(port: int) -> Runtime:
 
 class Client:
     """
-    A class which wraps the :class:`temporalio.client.Client` class
+    A class which wraps the :class:`temporalio.client.Client` class with reconnect logic.
     """
-    
+
     _is_stop_token_refresh = False
 
     @classmethod
-    def __del__(self):
-        self._is_stop_token_refresh = True
+    def __del__(cls):
+        cls._is_stop_token_refresh = True
 
     @classmethod
-    async def update_rpc_metadata_loop(self, client_opt, rpc_metadata):
+    async def reconnect_loop(cls):
         """
-        Periodically update the rpc_metadata headers
+        Reconnects to the Temporal server periodically when the token expires.
         """
-        while not self._is_stop_token_refresh:
-            # By default, refresh every 55 minutes. This is because Google OAuth
-            # tokens expire after 60 minutes.
-            await asyncio.sleep(3300)
-            auth_header_provider = AuthHeaderProvider(client_opt.auth)
-            rpc_metadata.update(auth_header_provider.get_headers())
+        while not cls._is_stop_token_refresh:
+            try:
+                await asyncio.sleep(3300)  # Refresh tokens every ~55 minutes (OAuth tokens last 60 min)
+                logging.info("Refreshing token and reconnecting to Temporal server...")
+                await cls._reconnect()
+            except Exception as e:
+                logging.error(f"Failed to reconnect to Temporal server: {e}")
+                await asyncio.sleep(60)  # Backoff before retrying
```

Comment:
polish: Perhaps add exponential backoff?

e.g.
```python
retries = 0
try:
...
except:
    retries += 1
    await asyncio.sleep(min(10 * 2**retries, 300))
else:
    retries = 0
```

Reply:
Good idea, added exponential backoff to retries. In doing so I realized another bug in which if the attempt to re-initialize the connection fails, it will sleep for the full 55 minutes (token refresh interval) before re-initializing the connection. I've since changed the order of this logic so it re-initializes the connection first before sleeping for 55 minutes.